extensions/wikia/AdEngine/js/slot/service/slotRegistry.js

```diff
@@ -40,9 +41,10 @@ define('ext.wikia.adEngine.slot.service.slotRegistry',  [
 		}
 	}
 
-	function setState(slotName, state) {
+	function setState(slotName, state, status) {
 		var slot = get(slotName);
 		slotStates[slotName] = state;
+		slotStatuses[slotName] = status;
```

Comment:
Can we have a separate `setStatus` method?

Reply:
Or rename `setState` to sth not tightly connected with state.

Reply:
Status can be set only when enabling/disabling slot. I resolve all comments here.